oidc_provider/tests/cases/test_authorize_endpoint.py

```diff
@@ -497,6 +501,46 @@ def test_strip_prompt_login(self):
         self.assertIn('none', strip_prompt_login(path3))
         self.assertNotIn('login', strip_prompt_login(path3))
 
+    def test_public_client_require_pkce(self):
+        """
+        Test that a public client with require_pkce set rejects requests without the code
+        challenge.
+        """
+
+        data = {
+            'client_id': self.client_public_pkce.client_id,
+            'response_type': 'code',
+            'redirect_uri': self.client_public_pkce.default_redirect_uri,
+            'scope': 'openid email',
+            'state': self.state,
+            'prompt': 'consent'
+        }
+
+        response = self._auth_request('get', data, is_user_authenticated=True)
+
+        self.assertIn('invalid_request', response['Location'])
```

Comment:
I think it would be good to show the "good path" version of this test too, where the code challenge is included. Unless I overlooked it in the earlier tests that show here.

Reply:
It's done implicitly in some of the other tests, for example [here](https://github.com/catalyst/django-oidc-provider/blob/2da8a3257f00c91c0ac7111e76c5ee05a707b1a1/oidc_provider/tests/cases/test_authorize_endpoint.py#L143-L178).

I agree adding a test to explicitly check this wouldn't be a bad idea.